### PR TITLE
Added support for OpenWeatherMap Daily Forecast

### PR DESCRIPTION
**Pull request by thomasfredericks**
Fixes #\<GitHub-issue-number\>.

Make sure all boxes are checked (add x inside the brackets) when you submit your contribution, remove this sentence before doing so.

- [ ] This PR is compliant with the [other contributing guidelines](https://github.com/ThingPulse/esp8266-weather-station/blob/master/CONTRIBUTING.md) as well (if not, please describe why).
- [ ] I have thoroughly tested my contribution.
- [ ] Should this code require changes to documentation I will contribute those to [https://github.com/ThingPulse/docs](https://github.com/ThingPulse/docs).

\<Description of and rationale behind this PR\>

### CODE REVIEW

**Comment by marcelstoer** [comment]
The 16 days/daily forecast is not available in the free tier: https://openweathermap.org/price. Therefore, I wouldn't want to have this function available in the library. We would get too many support requests from users if the find out the advertised function doesn't work.

@squix78 what's your take on this?

**Comment by squix78** [comment]
What would be the effect if somebody needs it and is willing to pay for it? They would most probably start a fork of this library and that can't be in the best interest of the library either. I would rather go for including it but stating clearly that we can't give support for it. Maybe we could add a matrix for the different services, just like the price matrix you linked to?

**Comment by thomasfredericks** [comment]
This is a paid service? Works for me on the free subscription. Weird.

**Comment by marcelstoer** [comment]
IIRC they had announced a several months long free-for-all period. Hence, it may very well be that it'll work for you for a few more weeks. I also remember a forum post where they made it very clear that this API will not be free.

We won't be able to support this API in the long run as we've only got free tier tokens. Unless OWM would issue free developer tokens for us we won't be able to test it.

**Comment by thomasfredericks** [comment]
Well, this is really I feature I want. The **regular** forecast is too detailed and does not really summarize a day. The **daily** forecast could be constructed from the **regular** one.

File: src/OpenWeatherMapForecastDaily.cpp [hunk 1]
```diff
@@ -0,0 +1,374 @@
...
+      data[currentForecast].clouds = value.toInt();
+  } else if (currentKey == "speed") {
+      data[currentForecast].windSpeed = value.toFloat();
+  } else if (currentKey == "deg") {
+      data[currentForecast].windDeg = value.toInt();
+  } else if (currentKey == "rain") {
+      data[currentForecast].rain = value.toFloat();
+  } 
+  	/*
+  // {"dt":1527066000,  uint32_t observationTime;
```
Comment (marcelstoer) [inline src/OpenWeatherMapForecastDaily.cpp]: Please remove all the dead code here.

File: src/OpenWeatherMapForecastDaily.h [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+  private:
+    String currentKey;
+    String currentParent;
+    OpenWeatherMapForecastDailyData *data;
+    uint8_t weatherItemCounter = 0;
+    uint8_t maxForecasts;
+    int8_t currentForecast = -1; 
+    boolean metric = true;
+    String language = "en";
+    //uint8_t *allowedHours;
```
Comment (marcelstoer) [inline src/OpenWeatherMapForecastDaily.h]: Ditto

File: src/OpenWeatherMapForecastDaily.h [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+  public:
+    OpenWeatherMapForecastDaily();
+    uint8_t updateForecasts(OpenWeatherMapForecastDailyData *data, String appId, String location, uint8_t maxForecasts);
+    uint8_t updateForecastsById(OpenWeatherMapForecastDailyData *data, String appId, String locationId, uint8_t maxForecasts);
+
+    void setMetric(boolean metric) { this->metric = metric; }
+    boolean isMetric() { return this->metric; }
+    void setLanguage(String language) { this->language = language; }
+    String getLanguage() { return this->language; }
+    /*void setAllowedHours(uint8_t *allowedHours, uint8_t allowedHoursCount) {
```
Comment (marcelstoer) [inline src/OpenWeatherMapForecastDaily.h]: Ditto

**Comment by marcelstoer** [comment]
OpenWeatherMap issued a developer token for ThingPulse that grants access to the 16 days/daily forecast API. Hence, we're now in a position to test and support this function now and in the future.

**Comment by marcelstoer** [comment]
Waiting for the copy-paste vs. refactoring discussion in https://github.com/ThingPulse/esp8266-weather-station/pull/183#issuecomment-623130626.

**Comment by mattblovell** [comment]
Per this page:

https://openweathermap.org/api/one-call-api

OpenWeather now provides daily forecasts (7 days, with high and low temperature for the day) as part of their free tier.

**Comment by marcelstoer** [comment]
One-call support was added in August, see #183.

**Comment by mattblovell** [comment]
> One-call support was added in August, see #183.

Ah, thanks!  I think the version I installed via Arduino's "Manage Libraries" must not be the latest.

**Comment by marcelstoer** [comment]
It is. That feature hasn't been released yet but it's on `master`.

**Comment by SillyHats** [comment]
If I import the ESP8266 Weather Station library downloaded from github into Arduino, how can I enable daily forecasts instead of hourly in esp8266-weather-station-color?

> It is. That feature hasn't been released yet but it's on `master`.

**Comment by marcelstoer** [comment]
As this code hasn't been merged into _this_ repository (i.e. ThingPulsees) you would have to get it from @thomasfredericks' repo.

**Pull request closed without merging**